File: languages/python/bloock/entity/encryption/aes_encrypter.py
import bloock._bridge.proto.encryption_entities_pb2 as proto
from bloock.entity.encryption.encrypter import Encrypter
from bloock.entity.encryption.encrypter_args import EncrypterArgs


class AesEncrypter(Encrypter):
    def __init__(self, args: EncrypterArgs) -> None:
        super().__init__(alg=proto.A256GCM, args=args)

    def to_proto(self) -> proto.Encrypter:
        local_key = None
        if self.args.local_key is not None:
            local_key = self.args.local_key.to_proto()

        managed_key = None
        if self.args.managed_key is not None:
            managed_key = self.args.managed_key.to_proto()

        return proto.Encrypter(
            alg=self.alg,
            local_key=local_key,
            managed_key=managed_key,
        )

File: languages/php/Bloock/Entity/Encryption/AesEncrypter.php
<?php

namespace Bloock\Entity\Encryption;

class AesEncrypter implements Encrypter
{
    public int $alg;
    public EncrypterArgs $args;

    public function __construct(EncrypterArgs $args)
    {
        $this->alg = \Bloock\EncryptionAlg::A256GCM;
        $this->args = $args;
    }

    public function toProto(): \Bloock\Encrypter
    {
        $p = new \Bloock\Encrypter();
        $p->setAlg($this->alg);

        if ($this->args->localKey != null) {
            $p->setLocalKey($this->args->localKey->toProto());
        }

        if ($this->args->managedKey != null) {
            $p->setManagedKey($this->args->managedKey->toProto());
        }

        return $p;
    }
}